tests/conftest.py

```diff
@@ -251,8 +252,17 @@ def output_directory(tmp_path: Path) -> Path:
 
 
 @fixture
-def application_pid(in_container: bool, application_process: subprocess.Popen, application_docker_container: Container):
-    return application_docker_container.attrs["State"]["Pid"] if in_container else application_process.pid
+def application_pid(
+    in_container: bool, application_process: subprocess.Popen, application_docker_container: Container
+) -> int:
+    pid = application_docker_container.attrs["State"]["Pid"] if in_container else application_process.pid
+
+    # Application might be run using "sh -c ...", we detect the case and return the "real" application pid
```

Comment:
Eh, why do you need this patch? To find the inner Java? Maybe `pgrep java` inside the container in your specific test?

Reply:
Meh, it's more ugly IMO, why not just fixing it that way?

Reply:
Because this solution is ugly as well and introduces unwanted logic into this previously-simple fixture. But fuck it